utility/confdService.py

```diff
@@ -43,70 +47,91 @@ def __init__(self):
         path = '{}/restconf'.format(self.confd_prefix)
         response = requests.get(path, auth=(self.credentials[0], self.credentials[1]), headers=confd_headers)
 
         return response
 
-    def get_module(self, module_key: str):
+    def get_module(self, module_key: str) -> requests.Response:
         self.LOGGER.debug('Sending GET request to the module {}'.format(module_key))
         path = '{}/restconf/data/yang-catalog:catalog/modules/module={}'.format(self.confd_prefix, module_key)
         modules_data = requests.get(path, auth=(self.credentials[0], self.credentials[1]), headers=confd_headers)
 
         return modules_data
 
-    def get_catalog_data(self):
+    def get_catalog_data(self) -> requests.Response:
         self.LOGGER.debug('Sending GET request to get all the catalog data')
         path = '{}/restconf/data/yang-catalog:catalog'.format(self.confd_prefix)
         catalog_data = requests.get(path, auth=(self.credentials[0], self.credentials[1]), headers=confd_headers)
 
         return catalog_data
 
-    def patch_module(self, module_key: str, new_data: str):
+    def patch_module(self, module: dict) -> requests.Response:
+        module_key = self.create_module_key(module)
         self.LOGGER.debug('Sending PATCH request to the module {}'.format(module_key))
+        patch_data = {'yang-catalog:module': module}
+        patch_json = json.dumps(patch_data)
         path = '{}/restconf/data/yang-catalog:catalog/modules/module={}'.format(self.confd_prefix, module_key)
-        response = requests.patch(path, new_data, auth=(self.credentials[0], self.credentials[1]), headers=confd_headers)
-
-        return response
-
-    def patch_modules(self, new_data: str):
-        self.LOGGER.debug('Sending PATCH request to patch multiple modules')
-        path = '{}/restconf/data/yang-catalog:catalog/modules/'.format(self.confd_prefix)
-        response = requests.patch(path, new_data, auth=(self.credentials[0], self.credentials[1]), headers=confd_headers)
+        response = requests.patch(path, patch_json, auth=(self.credentials[0], self.credentials[1]), headers=confd_headers)            
 
         return response
 
-    def patch_vendors(self, new_data: str):
-        self.LOGGER.debug('Sending PATCH request to patch multiple vendors')
-        path = '{}/restconf/data/yang-catalog:catalog/vendors/'.format(self.confd_prefix)
-        response = requests.patch(path, new_data, auth=(self.credentials[0], self.credentials[1]), headers=confd_headers)
-
-        return response
-
-    def delete_module(self, module_key: str):
+    def _patch(self, data: list, type: str, log_file: str):
+        errors = False
+        chunk_size = 500
+        chunks = [data[i:i + chunk_size] for i in range(int(len(data)/chunk_size) + 1)]
+        path = '{}/restconf/data/yang-catalog:catalog/{}/'.format(self.confd_prefix, type)
+        for chunk in chunks:
+            self.LOGGER.debug('Sending PATCH request to patch multiple modules')
```

Comment:
Move this log out of for loop.
Add new log in for loop which will l log something like:
```
Processing chunk 2 out of X
```